volttron/platform/agent/base_historian.py

```diff
@@ -1231,9 +1231,9 @@ def report_handled(self, record):
         """
         if isinstance(record, list):
             for x in record:
-                self._successful_published.add(x['_id'])
+                self._successful_published.add((x['timestamp'], x['topic']))
```

Comment:
I don't think this needed to change in the thoughts I had...I dont' think this will be good for cache as the two columns that are mentioned in the deletion is not (nor should they) be indexed.

Reply:
I think that the data should be removed from the database right as a duplicate is detected rather than waiting for a record to be passed.

Reply:
I see what you mean. When I saw that the field `_successful_published` was being used to remove things from the cache, I went in the direction of removing dupes under the `remove_successfully_published` method. Will change this and remove the dupes at the point of discovery.